ModuleFourTasks/Task3/UserTaskService.cs

```diff
@@ -12,25 +12,29 @@ public UserTaskService(IUserDao userDao)
             _userDao = userDao;
         }
 
-        public int AddTaskForUser(int userId, UserTask task)
+        public void AddTaskForUser(int userId, UserTask task)
         {
             if (userId < 0)
-                return -1;
+            {
+                throw new ArgumentException("Invalid userId");
+            }
 
             var user = _userDao.GetUser(userId);
             if (user == null)
-                return -2;
+            {
+                throw new ArgumentException("User not found");
+            }
 
             var tasks = user.Tasks;
             foreach (var t in tasks)
             {
                 if (string.Equals(task.Description, t.Description, StringComparison.OrdinalIgnoreCase))
-                    return -3;
+                {
+                    throw new ArgumentException("The task already exists");
```

Comment:
You can also use `InvalidOperationException`

Reply:
Fixed in cbd4ea3.